src/Analyzers/CSharp/Tests/AddExplicitCast/AddExplicitCastTests.cs

```diff
@@ -3076,6 +3074,32 @@ public void M(object o)
 
         if (array[(int)o] > 0) {}
     }
+}");
+        }
+
+        [Fact, WorkItem(56141, "https://github.com/dotnet/roslyn/issues/56141")]
```

Comment:
I previously discussed this issue with @RikkiGibson, and he thinks this should be handled on the IDE side rather than changing diagnostic span on compiler side.

Reply:
The discord logs from March 2022:

@Youssef1313 said, regarding https://github.com/dotnet/roslyn/issues/56141#issuecomment-912417482:
> Do you think the compiler behavior should be changed (ie, the diagnostic span), which will likely fix the IDE behavior for free, or do you think the fix should just go into the IDE?
> If the compiler, consider updating the issue label. Thanks!

@RikkiGibson said:
> I'm weighing it. I think the diagnostic span is probably the whole thing because within we're evaluating a + b whose result type is double then trying to convert implicitly to int.
> I think it will be lower-risk to update this in the IDE.